cmd/xl-v1-utils.go

```diff
@@ -357,6 +357,23 @@ func shuffleDisks(disks []StorageAPI, distribution []int) (shuffledDisks []Stora
 	return shuffledDisks
 }
 
+// evalDisks - returns a new slice of disks where nil is set if
+// the correspond error in errs slice is not nil
+func evalDisks(disks []StorageAPI, errs []error) []StorageAPI {
+	if len(errs) != len(disks) {
```

Comment:
This should never happen, returning a nil should accompany a log message here with errorIf() indicating something really unexpected has happened.